lib/helper/Graphql.js

```diff
@@ -0,0 +1,177 @@
+ * ```
+ *
+ * ## Access From Helpers
+ *
+ * Send GraphQL requests by accessing `_executeQuery` method:
+ *
+ * ```js
+ * this.helpers['GraphQL']._executeQuery({
+ *    url,
+ *    data,
+ * });
+ * ```
+ *
+ * ## Methods
+ */
+class GraphQL extends Helper {
+  constructor(config) {
+    super(config);
+    axios = requireg('axios');
+    this.options = {
+      timeout: 10000,
+      defaultHeaders: {},
+      endpoint: '',
+    };
+    this.options = Object.assign(this.options, config);
+    headers = Object.assign({}, this.options.defaultHeaders);
+    axios.defaults.headers = this.options.defaultHeaders;
+  }
+
+  static _checkRequirements() {
+    try {
+      requireg('axios');
+    } catch (e) {
+      return ['axios'];
+    }
+  }
+
+  /**
+   * Executes query via axios call
+   *
+   * @param {object} request
+   */
+  async _executeQuery(request) {
+    axios.defaults.timeout = request.timeout || this.options.timeout;
+
+    if (headers && headers.auth) {
+      request.auth = headers.auth;
+    }
+
+    request.headers = Object.assign(request.headers, {
+      'Content-Type': 'application/json',
+    });
+
+    if (this.config.onRequest) {
+      await this.config.onRequest(request);
+    }
+
+    this.debugSection('Request', JSON.stringify(request));
+
+    let response;
+    try {
+      response = await axios(request);
+    } catch (err) {
+      if (!err.response) throw err;
+      this.debugSection(
+        'Response',
+        `Response error. Status code: ${err.response.status}`,
+      );
+      response = err.response;
+    }
+    this.debugSection('Response', JSON.stringify(response.data));
+    return response;
+  }
+
+  /**
+   * Prepares request for axios call
+   *
+   * @param {object} operation
+   * @param {object} headers
+   */
+  prepareGraphQLRequest(operation, headers) {
+    const request = {
+      baseURL: this.options.endpoint,
+      method: 'POST',
+      data: JSON.stringify(operation),
+      headers,
+    };
+    return request;
+  }
+
+  /**
+   * Send query to GraphQL endpoint over http
+   *
+   * ```js
+   * I.sendQuery({
+   *    query: 'query { user { name email }}',
+   * });
+   * ```
+   *
+   * @param {object} operation
+   * @param {object} headers
+   */
+  async sendQuery(operation, headers = {}) {
+    const request = this.prepareGraphQLRequest(operation, headers);
+    return this._executeQuery(request);
+  }
+
+  /**
+   * Send query to GraphQL endpoint over http
+   *
+   * ```js
+   * I.sendMutation({
+   *    query: `
+   *       mutation createUser($user: UserInput!) {
+   *          createUser(user: $user) {
+   *            id
+   *            name
+   *            email
+   *          }
+   *        }
+   *    `,
+   *   variables: {
+   *     user: {
+   *       name: 'John Doe',
+   *       email: 'john@xmail.com'
+   *     }
+   *   },
+   * });
+   * ```
+   *
+   * @param {object} operation
+   * @param {object} headers
+   */
+  async sendMutation(operation, headers = {}) {
```

Comment:
Sure, that sounds good. Then perhaps there also should be a third argument like
`sendMutation(query, variables, options, headers)`
so that the user can attach additional attributes to the operation if need be? What do you think about that?